### PR TITLE
refactor: switch typeof undefined checks with strict comparison of the value undefined

### PR DESCRIPTION
**Pull request by TkDodo**

### CODE REVIEW

File: packages/query-core/src/queryClient.ts [hunk 1]
```diff
@@ -511,18 +511,15 @@ export class QueryClient {
...
-    if (typeof defaultedOptions.throwOnError === 'undefined') {
+    if (defaultedOptions.throwOnError === undefined) {
       defaultedOptions.throwOnError = !!defaultedOptions.suspense
     }
 
-    if (
-      typeof defaultedOptions.networkMode === 'undefined' &&
-      defaultedOptions.persister
-    ) {
+    if (!defaultedOptions.networkMode && defaultedOptions.persister) {
```
Comment (TkDodo) [inline packages/query-core/src/queryClient.ts:522]: `networkMode` is a string literal type so doing the simplified `!` check should be safe. We can't do that for the above booleans obviously

File: packages/query-core/src/utils.ts [hunk 2]
```diff
@@ -122,10 +122,7 @@ export function matchQuery(
     return false
   }
 
-  if (
-    typeof fetchStatus !== 'undefined' &&
-    fetchStatus !== query.state.fetchStatus
-  ) {
+  if (fetchStatus && fetchStatus !== query.state.fetchStatus) {
```
Comment (TkDodo) [inline packages/query-core/src/utils.ts:125]: same here: `fetchStatus` is not a boolean, we just need to check if it was passed

**Comment by SergioMorchon** [comment]
It may be handy to use a more semantic utility with a positive naming convention.
Something like
```typescript
const isDefined = <Value>(value: Value | null | undefined): value is Value =>
  value !== null && value !== undefined;
```

**Comment by TkDodo** [comment]
We can't check for null because data can be null - it just can't be undefined

**Pull request merged**